File: Data Structures & Algorithms/invert-a-binary-tree/submission-2.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public TreeNode InvertTree(TreeNode root) {
        if (root == null) return null;
        var stack = new Stack<TreeNode>();
        stack.Push(root);

        while (stack.Count > 0)
        {
            var node = stack.Pop();

            var temp = node.right;
            node.right = node.left;
            node.left = temp;

            if (node.left != null) stack.Push(node.left);
            if (node.right != null) stack.Push(node.right);
        }
        return root;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        stack = [(root, root.val)]
        result = 0

        while stack:
            cur_node, max_so_far = stack.pop()

            if cur_node.val >= max_so_far:
                result += 1

            new_max_so_far = max(max_so_far, cur_node.val)
            if cur_node.left is not None:
                stack.append((cur_node.left, new_max_so_far))
            if cur_node.right is not None:
                stack.append((cur_node.right, new_max_so_far))

        return result

        